interface/src/app/node/node.js

```diff
@@ -300,4 +309,123 @@ angular.module('adage.node', [
       };
     }
   };
-}]);
+}])
+
+.directive('enrichedGenesets', ['MathFuncts', 'PickledGenesetsService', '$log',
+  function(MathFuncts, PickledGenesetsService, $log) {
+    return {
+      templateUrl: 'node/enriched_genesets.tpl.html',
+      restrict: 'E',
+      scope: {
+        organism: '=',
+        hWGenes: '=' // hWGenes is an array of high-weight Genes
+      },
+      link: function($scope) {
+        $scope.queryStatus = 'Connecting to the server ...';
+
+        var pValueCutoff = 0.05;
+        var pValueSigDigits = 3;
+
+        // This is an object, where each key is the geneset ID, and each value
+        // is an array of the genes that this geneset contains.
+        var genesetGenes = {};
+
+        // We want to fill out this array with the genesets that score a
+        // p-value of less than the cutoff in the hypergeometric test.
+        // Each geneset in this array will be an object containing just the
+        // desired information.
+        var relevantGenesetArray = [];
+
+        // This is the main function that calculates the geneset enrichments.
+        // It calculates the enrichment for each geneset that has genes
+        // also present in the node high weight genes, and pushes that geneset
+        // into the releventGenesetArray.
+        var calculateEnrichments = function(geneGenesets, allGenesetInfo,
+                                            totalGeneNum, cutoff) {
+          var N = totalGeneNum;
+
+          // This will be the number of genes from the high weight gene list
+          // that are also present in any of the genesets that were returned
+          var m = 0;
+
+          // Fill out the genesetGenes object
+          for (var i = 0; i < $scope.hWGenes.length; i++) {
+            var genesetList = null;
+            var geneEntrezID = $scope.hWGenes[i].entrezID;
+
+            if (geneGenesets.hasOwnProperty(geneEntrezID)) {
```

Comment:
Took me a minute to figure out that `geneGenesets` was not a typo of `genesetGenes`. I now see the symmetry... think I'm okay with it, but if I get confused again might recommend changing one to be a more distinct name.